agent/task_context.go

```diff
@@ -288,7 +289,9 @@ func (tc *taskContext) getTimeoutType() globals.TimeoutType {
 func (tc *taskContext) getExecTimeout() time.Duration {
 	tc.RLock()
 	defer tc.RUnlock()
-	if dynamicTimeout := tc.taskConfig.GetExecTimeout(); dynamicTimeout > 0 {
+	// TODO DEVPROD-11204: Delete override to max exec timeout once this is enforced in project parsing
```

Comment:
I imagine we could just keep this after DEVPROD-11204 is done, even if the project validator errors on it. We have to store the max exec timeout for timeout.update anyways, so enforcing it on the actual timeout mechanism is a nice guardrail. It also means that if we change the max exec timeout in the future, old tasks can't exceed the current max exec timeout.